packages/ketcher-react/src/script/editor/tool/helper/dropAndMerge.ts

```diff
@@ -47,6 +48,14 @@ export function dropAndMerge(
     dropItemAction = dropItemAction.mergeWith(action)
   }
 
+  const bonds = editor.selection()?.bonds ?? []
+  for (const bondId of bonds) {
+    const rebond = restruct.bonds.get(bondId)
+    if (rebond) {
+      ReBond.bondRecalc(rebond, restruct, editor.options())
```

Comment:
Here we add bond recalculation b/c without it the position values for halfBonds do not update